lib/createWorkers.js

```diff
@@ -158,21 +200,26 @@ function createWorkers(handlers, config, options = {}) {
    */
   function _getMessageConsumer(channel, handle, validate) {
     return function* _consumeMessage(message) {
-      logger.debug({ message }, '[worker#listen] received message');
+      try {
+        logger.debug({ message }, '[worker#listen] received message');
 
-      const content = _parseMessage(message);
-      if (!content) return channel.ack(message);
+        const content = _parseMessage(message);
+        if (!content) return channel.ack(message);
 
-      const validatedContent = _validateMessage(validate, content);
-      if (_.isError(validatedContent)) return channel.ack(message);
+        const validatedContent = _validateMessage(validate, content);
+        if (_.isError(validatedContent)) return channel.ack(message);
 
-      const handleSuccess = yield _handleMessage(
-        handle,
-        validatedContent || content,
-        message.fields
-      );
-      if (!handleSuccess) return channel.nack(message);
-      return channel.ack(message);
+        const handleSuccess = yield _handleMessage(
+          handle,
+          validatedContent || content,
+          message.fields
+        );
+        if (!handleSuccess) return channel.nack(message);
+        return channel.ack(message);
+      } catch (err) {
+        logger.error({ err }, '[worker#listen] failed to ack/nack');
```

Comment:
Add the message and/or the routing key to be able to debug more easily the slowness problem?

Reply:
done